content/en/blog/extend-falco-inputs-with-a-plugin-distribute/index.md

```diff
@@ -0,0 +1,495 @@
+  * push the plugin
+  * push the rules with a dependency to the plugin
+
+```yaml
+jobs:
+  publish-oci-artifacts:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Falcoctl Repo
+        uses: actions/checkout@v3
+        with:
+          repository: falcosecurity/falcoctl
+          ref: main
+          path: tools/falcoctl
+      - name: Setup Golang
+        uses: actions/setup-go@v4
+        with:
+          go-version: '^1.20'
+          cache-dependency-path: tools/falcoctl/go.sum
+      - name: Build falcoctl
+        run: make
+        working-directory: tools/falcoctl
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: plugin
+      - name: Build the plugin
+        run: make build
+        working-directory: plugin
+      - id: StringRepoName
+        uses: ASzc/change-string-case-action@v5
+        with:
+          string: ${{ github.repository }}
+      - name: Upload OCI artifacts to GitHub packages
+        run: |
+              MAJOR=$(echo ${{ github.ref_name }} | cut -f1 -d".")
+              MINOR=$(echo ${{ github.ref_name }} | cut -f1,2 -d".")
+              DIR=$(pwd)
+
+              cd plugin/
+              $DIR/tools/falcoctl/falcoctl registry push \
+              ${{ env.OCI_REGISTRY }}/${{ steps.StringRepoName.outputs.lowercase }}/plugin/${{ env.PLUGIN_NAME }}:${{ github.ref_name }} \
+              --config /dev/null \
+              --type plugin \
+              --version "${{ github.ref_name }}" \
+              --tag latest --tag $MAJOR --tag $MINOR \
+              --platform linux/amd64 \
+              --requires plugin_api_version:2.0.0 \
+              --depends-on ${{ env.PLUGIN_NAME }}-rules:${{ github.ref_name }} \
+              --name ${{ env.PLUGIN_NAME }} \
+              lib${{ env.PLUGIN_NAME }}.so
+    
+              cd rules/
+              $DIR/tools/falcoctl/falcoctl registry push \
+              ${{ env.OCI_REGISTRY }}/${{ steps.StringRepoName.outputs.lowercase }}/ruleset/${{ env.PLUGIN_NAME }}:${{ github.ref_name }} \
+              --config /dev/null \
+              --type rulesfile \
+              --version "${{ github.ref_name }}" \
+              --tag latest --tag $MAJOR --tag $MINOR \
+              --depends-on ${{ env.PLUGIN_NAME }}:${{ github.ref_name }} \
+              --name ${{ env.PLUGIN_NAME }}-rules \
+              ${{ env.PLUGIN_NAME }}_rules.yaml
+        env:
+          FALCOCTL_REGISTRY_AUTH_BASIC: ${{ env.OCI_REGISTRY }},${{ github.repository_owner }},${{ secrets.GITHUB_TOKEN }}
+```
+
+#### Create the release
+
+At the same time we publish new artifacts, we'll also create a release with an automatic changelog. It's done by [`Goreleaser`](https://goreleaser.com/). This section is out of context of this blog post, it's a classic with Go programs, just take care to have a correct `.goreleaser` as explained [here]({{< ref "/blog/extend-falco-inputs-with-a-plugin-distribute##goreleaser" >}}).
```

Comment:
This sentence is long and weird. I can't follow what you are trying to say. It is out of scope, but you are adding it. It is also a simple config, which would be easy to explain. I think that a quick explanation paragraph would be easier than this entire text to say that it is out of scope.

Reply:
I'd rather say something like 'out of scope', not out of context.

What about:
`GoReleaser` can automatically generate a _Changelog_ at the same time we publish the new artifacts. This step isn't imperative to generate the OCI artifacts but it's a good practice among Go developers. To achieve that, make sure to have a correct `.goreleaser` file as explained here.

Reply:
I changed for Vic's version. Thanks.